compiler/rustc_mir_transform/src/lib.rs

```diff
@@ -593,11 +593,11 @@ fn run_optimization_passes<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
             &multiple_return_terminators::MultipleReturnTerminators,
             &instsimplify::InstSimplify,
             &simplify::SimplifyLocals::BeforeConstProp,
-            &dead_store_elimination::DeadStoreElimination::Initial,
             &gvn::GVN,
             &simplify::SimplifyLocals::AfterGVN,
             &dataflow_const_prop::DataflowConstProp,
             &single_use_consts::SingleUseConsts,
+            &dead_store_elimination::DeadStoreElimination::Initial,
```

Comment:
I'd rather see `SingleUseConsts` run multiple times than move DSE.
Having DSE run early is interesting to reduce work and create additional opportunities for GVN.
Furthermore, SingleUseConsts should be cheap enough to run multiple times, doesn't it?

Reply:
:thinking: I've tried to apply this concept, and it seems to have fixed a FIXME in the SingleUseConsts test?